mmeval/metrics/dota_map.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import logging
+import numpy as np
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+from mmeval.metrics.voc_map import VOCMeanAP
+from .utils.bbox_overlaps_rotated import (calculate_bboxes_area_rotated,
+                                          qbox_to_rbox)
+
+logger = logging.getLogger(__name__)
+
+try:
+    # we prefer to use `bbox_iou_rotated` in mmcv to calculate ious
+    from mmcv.ops import box_iou_rotated
+    from torch import Tensor
+    HAS_MMCV = True
+except Exception as e:  # noqa F841
+    from .utils.bbox_overlaps_rotated import calculate_overlaps_rotated
+    HAS_MMCV = False
+    logger.debug(
+        'mmcv is not installed, calculating IoU of rotated bbox with OpenCV.')
+
+
+def filter_by_bboxes_area_rotated(bboxes: np.ndarray,
```

Comment:
This function has been already defined in `voc_map.py`.

Reply:
The similar function in voc_map.py is named `filter_by_bboxes_area`~